shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -595,15 +717,10 @@ - (FlutterPlatformViewController*)platformViewController {
 
 #pragma mark - Private methods
 
-- (FlutterViewController*)viewControllerForId:(uint64_t)viewId {
-  // TODO(dkwingsmt): The engine only supports single-view, therefore it
-  // only processes the default ID. After the engine supports multiple views,
-  // this method should be able to return the view for any IDs.
-  NSAssert(viewId == kFlutterDefaultViewId, @"Unexpected view ID %llu", viewId);
-  if (viewId == kFlutterDefaultViewId) {
-    return _viewController;
-  }
-  return nil;
+- (uint64_t)generateViewId {
+  uint64_t result = _nextViewId;
+  _nextViewId += 1;
+  return result;
```

Comment:
What happens if my app creates a view, transitions back to headless mode, then creates another view? Should we guarantee that at least one view has `kFlutterDefaultViewId` if the engine is not in headless mode?

For example, imagine my app is a popup from the menu bar. It starts in headless mode until the user opens the popup. The app returns to headless mode once the user closes the popup. (This is the Google One macOS app scenario)

Reply:
Overall, I think we should make fewer assumptions about what ID a view will have or get. Ideally, we make no assumptions at all about this. Once consumers of these APIs have migrated to the multi-view APIs they should really check what ID they got instead of relying on some default behavior.

Reply:
It depends on which API you're using. 

If you're using `setViewController`, then the resulting FVC will always have the default ID, since `setViewController` always operates the default view.

If you're using `addViewController`, then the following FVCs will never have the default ID again. This doesn't hurt, because since you're using a new API anyway, it should be safe to assume you're not relying on the default API suite.